plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StartTask.java

```diff
@@ -0,0 +1,354 @@
+        } else if (!getProject().getTasks().withType(War.class).isEmpty()) {
+            executor = warExecutor();
+        } else if (!getProject().getTasks().withType(Jar.class).isEmpty()) {
+            executor = jarExecutor();
+        } else {
+            throw new GradleException("Unable to determine type of the project, neither war or jar tasks were found.");
+        }
+
+        executor.withJVMArguments(extension.getJvmArguments());
+        executor.withArguments(extension.getArguments());
+
+        final SwarmProcess process;
+        try {
+
+            File tmp;
+            try {
+                tmp = Files.createTempFile("thorntail-process-file", null).toFile();
+                getLogger().info("Process file: " + tmp.toPath().toString());
+            } catch (IOException e) {
+                throw new GradleException("Error while creating process file");
+            }
+
+            process = executor.withDebug(extension.getDebugPort())
+                    .withProcessFile(tmp)
+                    .withProperties(extension.getProperties())
+                    .withProperties(ThorntailUtils.getPropertiesFromFile(extension.getPropertiesFile()))
+                    .withStdoutFile(extension.getStdoutFile() != null ? getProject().file(extension.getStdoutFile()).toPath() : null)
+                    .withStderrFile(extension.getStderrFile() != null ? getProject().file(extension.getStderrFile()).toPath() : null)
+                    .withEnvironment(extension.getEnvironment())
+                    .withEnvironment(ThorntailUtils.getPropertiesFromFile(extension.getEnvironmentFile()))
+                    .withWorkingDirectory(getProject().getProjectDir().toPath())
+                    .withProperty("remote.maven.repo",
+                            getProject().getRepositories().withType(MavenArtifactRepository.class).stream()
+                                    .map(mavenArtifactRepository -> mavenArtifactRepository.getUrl().toASCIIString())
+                                    .collect(Collectors.joining(",")))
+                    .execute();
+            registerCancellationListeners(process);
+
+            process.awaitReadiness(extension.getStartTimeout(), TimeUnit.SECONDS);
+
+            if (!process.isAlive()) {
+                throw new GradleException("Process failed to start");
+            }
+            if (process.getError() != null) {
+                throw new GradleException("Error starting process", process.getError());
+            }
+
+        } catch (IOException e) {
+            throw new GradleException("unable to execute", e);
+        } catch (InterruptedException e) {
+            throw new GradleException("Error waiting for deployment", e);
+        }
+
+        if (waitForProcess) {
+            try {
+                process.waitFor();
+            } catch (InterruptedException e) {
+                try {
+                    process.stop(10, TimeUnit.SECONDS);
```

Comment:
All should be fixed:
* Stop timeout is configurable, default is 120 seconds.
* Fixed the problem with the process stopping too slowly: https://github.com/TomasHofman/thorntail/commit/1c25dd6828540e7423c684aaf290667f1b5dcae6#diff-d252ac069893b056c7cc65cf320827b8L108

  The problem was that `SwarmProcess#stop()` first deleted the process file and than waited up to the `stopTimeout` for the process to terminate by itself, before it called `Process#destroy()`. You mentioned this was how it worked in the past - shutdown triggered by process file deletion. I deleted that waiting, since I believe that `Process#destroy()` should now be called straight away. `destroy()` now triggers the service shutdown.